app/models/notify_user/scheduler.rb

```diff
@@ -0,0 +1,46 @@
+module NotifyUser
+  class Scheduler
+    def self.schedule(notification)
+      new(notification).schedule
+    end
+
+    def initialize(notification)
+      @notification = notification
+    end
+
+    def schedule
+      notification.class.channels.each do |channel, options|
+        aggregator = Aggregator.new(notification, options[:aggregate_per])
+        return if aggregator.has_pending_deliveries?
+
+        # Create the delivery:
```

Comment:
I don't even really think the comment is necessary. Pretty clear what the line does.

Pretty sure I just left it there from my pseudo code so everything lined up nicely.